qualtran/bloqs/qsp/generalized_qsp.py

```diff
@@ -274,16 +340,51 @@ def decompose_from_registers(
             yield self.U.adjoint().on_registers(**quregs)
 
     def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
-        degree = len(self.P)
+        degree = len(self.P) - 1
 
-        counts = {(rotation, 1) for rotation in self.signal_rotations}
+        counts = set(Counter(self.signal_rotations).items())
 
         if degree > self.negative_power:
-            counts.add((self.U.controlled(control_values=[0]), degree - self.negative_power))
+            counts.add((Controlled(self.U, CtrlSpec(cvs=0)), degree - self.negative_power))
         elif self.negative_power > degree:
             counts.add((self.U.adjoint(), self.negative_power - degree))
 
         if self.negative_power > 0:
-            counts.add((self.U.adjoint().controlled(), min(degree, self.negative_power)))
+            counts.add((Controlled(self.U.adjoint(), CtrlSpec()), min(degree, self.negative_power)))
 
         return counts
+
+
+@bloq_example
+def _gqsp() -> GeneralizedQSP:
+    from qualtran.bloqs.for_testing.atom import TestGWRAtom
+
+    gqsp = GeneralizedQSP.from_qsp_polynomial(TestGWRAtom(), (0.5, 0.5))
+    return gqsp
+
+
+@bloq_example
+def _gqsp_with_negative_power() -> GeneralizedQSP:
+    from qualtran.bloqs.for_testing.atom import TestGWRAtom
+
+    gqsp_with_negative_power = GeneralizedQSP.from_qsp_polynomial(
+        TestGWRAtom(), (0.5, 0, 0.5), negative_power=1
+    )
+    return gqsp_with_negative_power
+
+
+@bloq_example
+def _gqsp_with_large_negative_power() -> GeneralizedQSP:
+    from qualtran.bloqs.for_testing.atom import TestGWRAtom
+
+    gqsp_with_large_negative_power = GeneralizedQSP.from_qsp_polynomial(
+        TestGWRAtom(), (0.5, 0, 0.5), negative_power=5
```

Comment:
Is `5` large enough? Does it support higher negative powers as well?

Reply:
It works for any negative power. When `negative_power` is more than the degree, the circuit has U^\dagger terms, which I wanted to showcase with the example.